src/test/kotlin/com/atlassian/performance/tools/report/distribution/DistributionComparatorTest.kt

```diff
@@ -0,0 +1,193 @@
+        }
+    }
+
+    @Test
+    fun shouldDescribePartialSlowdown() {
+        // given
+        val random = MersenneTwister(123)
+        val fastMode = NormalDistribution(random, 400.0, 30.0, DEFAULT_INVERSE_ABSOLUTE_ACCURACY)
+        val slowMode = NormalDistribution(random, 6000.0, 150.0, DEFAULT_INVERSE_ABSOLUTE_ACCURACY)
+        val baseline = fastMode.sample(500) + slowMode.sample(90)
+        val verticalShift = 800.0
+        val partBoundary = 0.40 * baseline.size
+        val experiment = baseline
+            .mapIndexed { index, latency ->
+                if (index < partBoundary) {
+                    latency + verticalShift
+                } else {
+                    latency
+                }
+            }
+            .toDoubleArray()
+
+        // when
+        val comparison = DistributionComparator.Builder(baseline, experiment).build().compare()
+
+        // then
+        plotQuantiles(baseline, experiment)
+        assertSoftly {
+            it.assertThat(comparison.experimentAbsoluteChange).`as`("absolute change").isEqualTo(66.57177057231809)
+            it.assertThat(comparison.experimentRelativeChange).`as`("relative change").isEqualTo(0.16145163153504116)
+            it.assertThat(comparison.isExperimentImproved).`as`("improved").isFalse
+            it.assertThat(comparison.isExperimentRegressed).`as`("regressed").isTrue
+        }
+    }
+
+    /**
+     * In a 51% slightly faster vs 49% much slower case, it should be a regression
+     */
+    @Test
+    @Ignore
+    fun shouldCareAboutHeightOfTheDifferences() {
+        // given
+        val random = MersenneTwister(123)
+        val fastMode = NormalDistribution(random, 400.0, 30.0, DEFAULT_INVERSE_ABSOLUTE_ACCURACY)
+        val baseline = fastMode.sample(500)
+        val slowdown = 800.0
+        val speedup = -200.0
+        val partBoundary = 0.51 * baseline.size
+        val experiment = baseline
+            .sorted()
+            .mapIndexed { index, latency ->
+                if (index < partBoundary) {
+                    latency + speedup
+                } else {
+                    latency + slowdown
+                }
+            }
+            .toDoubleArray()
+
+        // when
+        val comparison = DistributionComparator.Builder(baseline, experiment).build().compare()
+
+        // then
+        plotQuantiles(baseline, experiment)
+        assertSoftly {
+            it.assertThat(comparison.experimentAbsoluteChange).`as`("absolute change").isPositive
+            it.assertThat(comparison.experimentRelativeChange).`as`("relative change").isPositive
+            it.assertThat(comparison.isExperimentImproved).`as`("improvement").isFalse
+            it.assertThat(comparison.isExperimentRegressed).`as`("regressed").isTrue
+        }
+    }
+
+    @Test
+    fun shouldDetectImprovementWhenEveryPercentileBetter() {
+        // given
+        val baseline =
```

Comment:
Better to chop the line earlier e.g. `.readText().lines()` on a separate line, instead of hanging assignments.